ui/apps/platform/src/Containers/Vulnerabilities/utils/vulnerabilityUtils.ts

```diff
@@ -0,0 +1,47 @@
+import { severityRankings } from 'constants/vulnerabilities';
+import { VulnerabilitySeverity, isVulnerabilitySeverity } from 'types/cve.proto';
+
+/**
+ * Get the highest severity of any vulnerability in the component array.
+ */
+export function getHighestVulnerabilitySeverity(
+    vulnerabilities: { severity: string }[]
+): VulnerabilitySeverity {
+    let topSeverity: VulnerabilitySeverity = 'UNKNOWN_VULNERABILITY_SEVERITY';
+    vulnerabilities.forEach(({ severity }) => {
+        if (
+            isVulnerabilitySeverity(severity) &&
+            severityRankings[severity] > severityRankings[topSeverity]
+        ) {
+            topSeverity = severity;
+        }
+    });
+    return topSeverity;
+}
+
+/**
+ * Get whether or not any component has any fixable vulnerabilities.
+ */
+export function getAnyVulnerabilityIsFixable(
```

Comment:
Here is an alternative name which answers my instinctive question when reading at the call site:

```ts
const isFixable = getIsSomeVunlerabilityFixable(vulnerabilities);
```

In what linguists call free variation, another occurrence prompted me to comment:

```ts
const isAnyVulnFixable = getIsSomeVulnerabilityFixable(vulnerabilities); // rename variable?
```

Reply:
+1 on the function rename, I'll push that change.

For the variable naming, the context in which this is used is always for a _single_ CVE from what I can see. The concern for multiples is because of the context within a "component", so the single vulnerability may be fixable or not fixable in any number of components in the image, deployment, etc.

`isAnyVulnFixable` was derived from the fact that we need to check multiple vulnerability objects to determine this, due to the GraphQL structure, but in reality it is always the same single vulnerability. In this context, since `isAnyComponentFixable` doesn't really seem ideal either, I think `hasFixableVulnerabilities` could work. Or perhaps `isFixableInImage`, `isFixableInDeployment` depending on the context in which it is used.

Reply:
Aha, names like `isFixableInImage` or `isFixableInDeployment` will help future readers!